### ISSUE TITLE
privilege rights mapping issue

### ISSUE THREAD

**Issue by logicminds** [ayohrling/local_security_policy#14]
There seems to be an issue when using the privilege rights policy type when it comes to mapping users to sids. Puppet thinks that it needs to update something when a username is supplied instead of the sid.  We just need to change when we covert the sid to a human readable name so that puppet does not get confused.

```
Notice: /Stage[main]/Main/Local_security_policy[Log on as a service]/policy_valu
e: policy_value changed 'sshd_server,*S-1-5-80-0' to '*S-1-5-21-2779173042-37658
3331-3911003981-1003,*S-1-5-80-0'
Notice: Finished catalog run in 2.06 seconds
```

This occurs when the following code is applied

```
local_security_policy { 'Log on as a service':
  ensure         => 'present',
  policy_setting => 'SeServiceLogonRight',
  policy_type    => 'Privilege Rights',
  policy_value   => 'sshd_server,*S-1-5-80-0',
}
```

**Comment by logicminds** [comment]
This is most likely in the self.instances function

**Comment by ayohrling** [comment]
Right, I ran into this when the group/user value is not part of the array in the lsp ruby file for accounts. I think the issue is that to be able to enter them in human-readable they have to be converted, but when Puppet looks at the instances I couldn't find a way to convert the value on the fly (nor could Paul initially for that matter) so Puppet always reads non-standard entries as their UUIDs instead of the name.

**Comment by ayohrling** [comment]
Ideally instead of converting via the pre-populated array, that would be replaced with a Windows API call that would convert any value to the UUID

**Comment by logicminds** [comment]
Well this is actually converting the value but I think the individually parameter should have a munge block as well to convert the value.

https://github.com/ayohrling/local_security_policy/blob/master/lib/puppet/type/local_security_policy.rb#L102

https://github.com/ayohrling/local_security_policy/blob/master/lib/puppet/type/local_security_policy.rb#L65

The provider code self.instances should convert the value to human format https://github.com/ayohrling/local_security_policy/blob/master/lib/puppet/provider/local_security_policy/policy.rb#L82

Basically we have the human readable content (user name) and the computer generated guid which is dynamic.  But the user is able to supply both of these formats in a resource so we need to marshal the values back and forth. 

This would need to be able to convert everything to human format and then convert back to guid when applying the update.

**Comment by natemccurdy** [comment]
This is likely the same problem as issue #27

**Comment by rismoney** [comment]
Does this help:
https://github.com/puppetlabs/puppetlabs-acl/blob/master/lib/puppet/provider/acl/windows/base.rb#L64-L87

**Comment by rychannel** [comment]
Anyone come up with any work arounds for this?

**Comment by logicminds** [comment]
I am no longer involved nor have a need for this module.

**Comment by ayohrling** [comment]
Removed you as a collaborator. Thanks for all the prior support!

**Comment by elserhumano** [comment]
Hi ayohrling, do you've some update?

I contact you because I use your module to setup a user as "log on as a service", when I use the human format (username) work fine but apply every time, if I use the guid work well and apply only when made changes!

**Comment by rychannel** [comment]
I have a custom local group created and it runs in the same issue.  Just keeps reapplying

**Comment by elserhumano** [comment]
I solved it by putting the user's SID instead of the "human name", for example change "domain\user_of_domain" by the SID: *S-1445-22334-.... 
I don't know for what reason need to have the "*" at the begining!

**Comment by elserhumano** [comment]
asterisk

**Comment by rychannel** [comment]
My problem is that the group is a local group on each system, so the SID's would be different for each system

**Comment by elserhumano** [comment]
Hi, if you're in an Active Directory, I don't remember well but I believe that you can create a group in the domain and made a member of the local group, sorry in this part I don't remember well.

But if you can do this, you can put the local group in that domain group that cannot change the sid, and don't care about the sid of the local group, ask only for the sid of the domain group.

**Comment by rychannel** [comment]
That doesn't help me because I need to use a local group to keep access to specific users limited and I don't want to create a bunch of groups when I have puppet just created a local group on each server.

**Comment by AndyAdrian-OP** [comment]
@rychannel You should be able to use the local group name in your resource now, as of https://github.com/ayohrling/local_security_policy/pull/99.
